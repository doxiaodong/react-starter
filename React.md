## Base

* lint (eslint, prettier, airbnb)
* test (jest, enzyme)
* ci/husky (npm run lint && npm run test && npm run build)
* lock (package-lock.json/yarn.lock)
* git editorconfig (LF)

## Config

* dev-server (proxy)
* uglify (html, css, js -> babel-minify)
* merge
* tree-shaking
* code split
* postcss
* svg-split/iconfont
* env (dev, test, prod)
* cdn (publicPath)
* ssr (mobx redux)

## Code

* index.js
* api
* utils
* views
* ui-components
* store/stores (redux/mobx)
* style
* form
* dependencies
  - classnames
  - omit.js
  - moment
  - lodash-es
  - normalize.css

### Component
* form
  - https://github.com/foxhound87/mobx-react-form
* animate
  - https://github.com/react-component/animate
* hammerjs
  - https://github.com/JedWatson/react-hammerjs
