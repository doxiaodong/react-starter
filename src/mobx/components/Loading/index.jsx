import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Loading extends Component {
    render() {
        const { desc } = this.props;
        return (
            <div>
                {desc}
            </div>
        );
    }
}

Loading.propTypes = {
    desc: PropTypes.string
};

Loading.defaultProps = {
    desc: 'loading...'
};
