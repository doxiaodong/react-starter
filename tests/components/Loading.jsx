import React from 'react';
import { shallow, render } from 'enzyme';
// why need cheerio, see https://github.com/airbnb/enzyme/issues/1162
import cheerio from 'cheerio'
import Loading from 'components/Loading';

describe('Loading Component', () => {
    test('Snapshot', () => {
        const component = render(<Loading />);
        expect(cheerio.html(component)).toMatchSnapshot();
    });

    test('Enzyme', () => {
        const wrapper = shallow(<Loading />);
        expect(wrapper.find('div')).toHaveLength(1);
    });
});
