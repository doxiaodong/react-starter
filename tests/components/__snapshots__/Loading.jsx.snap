// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component Snapshot 1`] = `"<div>loading...</div>"`;
